### PR TITLE
(maint) Update to have version requirements in metadata.json

### PR DESCRIPTION
**Pull request by cyberious**

### CODE REVIEW

**Comment by MikaelSmith** [comment]
Technically it will run on 4.0.0, it just won't do anything.

File: metadata.json [hunk 1]
```diff
@@ -40,6 +40,12 @@
       ]
     }
   ],
+  "requirements": [
+    {
+      "name": "puppet",
+      "version_requirement": ">= 3.8.0 < 4.0.0"
```
Comment (ferventcoder) [inline metadata.json]: Does this version requirement apply to the master that I would install this on to later upgrade my agents?

Comment (ferventcoder) [inline metadata.json]: Because I'd just state `>= 3.8.0` if so.

Comment (MikaelSmith) [inline metadata.json]: Also, does it have an impact on acceptance testing, where the tests run `puppet agent` after upgrading?

**Comment by MikaelSmith** [comment]
Why is the forge claiming we have 7 warnings, when the lint run here doesn't say any?

**Comment by MikaelSmith** [comment]
ping @cyberious

**Comment by MikaelSmith** [comment]
Closed in favor of #47.

**Pull request closed without merging**